docs/core-concepts/core-sdk/cli/extensibility.md

```diff
@@ -0,0 +1,174 @@
+the tools and explain the scenarios that drive each of them. It will the outline how to consume the tools as well as 
+provide short notes on how to build both types of tools. 
+
+## Two different ways to extend CLI
+The CLI tools can be extended in two main ways:
+
+1. Via NuGet packages on a per-project basis
+2. Via the system's PATH
+
+The two extensibility mechanisms outlined above are not exclusive; you can use both or just one. Which one to pick 
+depends largely on what is the goal you are trying to achieve with your extension.
+
+## Per-project based extensibility
+Per-project tools are [portable console applications](../../app-types.md) that are distributed as NuGet packages. Tools are 
+only available in the context of the project that references them and for which they are restored; invocation outside 
+of the context of the project (e.g. outside of the directory that houses the project) will fail as the command will 
+not be able to be found.
+
+These tools are perfect for build servers as well, since nothing outside of `project.json` is needed. The build process 
+runs restore for the project it builds and tools will be available. Language projects, such as F#, are also in this 
+category; after all, each project can only be written in one specific language. 
+
+Finally, this extensibility model provides support for creation of tools that need access to the built output of the 
+project. For instance, various Razor view tools in [ASP.NET](https://www.asp.net/) MVC applications fall into this 
+category. 
+
+### Consuming per-project tools
+Consuming these tools requires you to add a `tools` node to your `project.json`. Inside the `tools` node, you reference
+the package in which the tool resides. After running `dotnet restore`, the tool and its dependencies are restored. 
+
+For tools that need to load the build output of the project for execution, there is usually another dependency which is 
+listed under the regular dependencies in the project file. This means that tools that load project's code have two 
+components: 
+
+1. The "tools" main invoker
+2. Any number of other tools that contain the logic to work with 
+
+Why two things? Tools that need to load the build output of a project need to have unified dependency graph with the 
+project they are working. By adding the dependency bit, we enable NuGet to resolve these dependencies as an unified 
+graph. The invoker is there because it needs to reason about the location as well as the frameworks of the dependency 
+tool. The invoker can accept all of the redirection arguments (`-c`, `-o`, `-b`) that the user specify and find the 
+dependency tool; it can also implement any policies for cases where multiple dependency tools exist for multiple 
+frameworks (i.e. does it run all of them, just one etc.) In general, logic can be shared between these two tools any way 
```

Comment:
that the user specify -> that the user specifies
find -> finds
i.e. -> that is 
missing comma before etc.